src/main/java/com/prowidesoftware/swift/io/AbstractReader.java

```diff
@@ -64,10 +68,16 @@ public AbstractReader(final String string) {
      *
      * @param stream input stream
      * @throws IllegalArgumentException if stream is null
+     * @deprecated use constructor signature with charset instead
      */
+    @ProwideDeprecated(phase2 = TargetYear.SRU2022, comment = "use constructor signature with charset instead")
     public AbstractReader(final InputStream stream) {
-        Validate.notNull(stream, "stream must not be null");
-        this.reader = new InputStreamReader(stream);
+        this(stream, null);
```

Comment:
(I guess the reason for this change is that SpotBugs complains about relying on the default charset)
I'd rather keep this method, not deprecated, with UTF-8 as charset. Many users don't have a clue on charset/encodings so I think it is a good idea to have a simple entry point for them. And UTF-8 seems like a good choice. Actually I thought it was Java default, but now I see it is the system's default.

Reply:
No problem.
I'll leave the new signatures with Charset but will remove deprecation annotations.